src/worker/vm/lazy.ts

```diff
@@ -14,38 +15,72 @@ import { clearError, processError } from '../../utils/db/error'
+    totalAttemptsToInitialize: number
 
     constructor() {
+        this.totalAttemptsToInitialize = 0
+        this.initVM()
+    }
+
+    private initVM() {
+        this.totalAttemptsToInitialize++
         this.resolveInternalVm = new Promise((resolve) => {
             this.initialize = async (hub: Hub, pluginConfig: PluginConfig, indexJs: string, logInfo = '') => {
-                try {
-                    const vm = await createPluginConfigVM(hub, pluginConfig, indexJs)
+                const createPluginLogEntry = async (
+                    message: string,
+                    logType = PluginLogEntryType.Info
+                ): Promise<void> => {
                     await hub.db.createPluginLogEntry(
                         pluginConfig,
                         PluginLogEntrySource.System,
-                        PluginLogEntryType.Info,
-                        `Plugin loaded (instance ID ${hub.instanceId}).`,
+                        logType,
+                        message,
                         hub.instanceId
                     )
+                }
+                try {
+                    const vm = await createPluginConfigVM(hub, pluginConfig, indexJs)
+                    await createPluginLogEntry(`Plugin loaded (instance ID ${hub.instanceId}).`)
                     status.info('🔌', `Loaded ${logInfo}`)
                     void clearError(hub, pluginConfig)
                     await this.inferPluginCapabilities(hub, pluginConfig, vm)
                     resolve(vm)
                 } catch (error) {
-                    await hub.db.createPluginLogEntry(
-                        pluginConfig,
-                        PluginLogEntrySource.System,
-                        PluginLogEntryType.Error,
-                        `Plugin failed to load and was disabled (instance ID ${hub.instanceId}).`,
-                        hub.instanceId
+                    const isRetryError = error instanceof RetryError
+                    status.warn('⚠️', error.message)
+                    if (isRetryError && this.totalAttemptsToInitialize < 15) {
+                        const nextRetryMs = 2 ** this.totalAttemptsToInitialize * 3000
+                        const nextRetrySeconds = `${Math.round(nextRetryMs / 1000)}s`
+                        status.warn(
+                            '⚠️',
+                            `Failed to load ${logInfo}. Retrying to initialize it in ${nextRetrySeconds}.`
+                        )
+                        await createPluginLogEntry(
+                            `Plugin failed to load but its initialization will be retried in ${nextRetrySeconds} (instance ID ${hub.instanceId}).`,
+                            PluginLogEntryType.Error
+                        )
+                        setTimeout(() => {
+                            this.initVM()
+                            void this.initialize?.(hub, pluginConfig, indexJs, logInfo)
+                        }, nextRetryMs)
```

Comment:
Q: what if the plugin is reloaded (e.g. after it's updated) before this timeout callback runs?

Reply:
It wouldn't impact the plugin because a new VM is created on reload anyway. 

However, we may indeed end up with a loose timeout. I'll store the timeout and clear it on teardown